smokey/src/main/java/frc/robot/subsystems/CameraSubsystem.java

```diff
@@ -10,44 +10,53 @@
 
 package frc.robot.subsystems;
 
-import edu.wpi.first.wpilibj2.command.CommandBase;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import frc.robot.control.SubsystemCollection;
+
+/**
+ * A class to encapsulate the camera subsystem
+ */
 public class CameraSubsystem extends SubsystemBase {
-  /** Creates a new ExampleSubsystem. */
-  public CameraSubsystem() {}
+  DrivetrainSubsystem drivetrainSubsystem;
+  private final double milisecondsInSeconds = 1000.0;
+  private final int defaultDoubleArraySize = 7;
+  private final int TimestampIndex = 6;
+  private final int botPositionXIndex = 0;
+  private final int botPositionYIndex = 2;
+  private final int botRotationIndex = 5;
 
   /**
-   * Example command factory method.
-   *
-   * @return a command
+   * a constructor for the camera subsystem class
+   * @param subsystems - the subsystem collection
    */
-  public CommandBase exampleMethodCommand() {
-    // Inline construction of command goes here.
-    // Subsystem::RunOnce implicitly requires `this` subsystem.
-    return runOnce(
-        () -> {
-          /* one-time action goes here */
-        });
+  public CameraSubsystem(SubsystemCollection subsystems) {
+    drivetrainSubsystem = subsystems.getDriveTrainSubsystem();
   }
 
   /**
-   * An example method querying a boolean state of the subsystem (for example, a digital sensor).
-   *
-   * @return value of some boolean subsystem state, such as a digital sensor.
+   * A method to run during periodic for the camera subsystem
+   * it reads tags and updates the estimated position in the drivetrain subsystem
    */
-  public boolean exampleCondition() {
-    // Query some boolean state, such as a digital sensor.
-    return false;
-  }
-
   @Override
   public void periodic() {
-    // This method will be called once per scheduler run
-  }
-
-  @Override
-  public void simulationPeriodic() {
-    // This method will be called once per scheduler run during simulation
+    NetworkTable table = NetworkTableInstance.getDefault().getTable("limelight");
+    double tagId = table.getEntry("tid").getDouble(0);
+    double[] botpose = table.getEntry("botpose").getDoubleArray(new double[defaultDoubleArraySize]);
+    Double timestamp = Timer.getFPGATimestamp() - (botpose[TimestampIndex]/milisecondsInSeconds);
+    Translation2d botTranslation = new Translation2d(botpose[botPositionXIndex], botpose[botPositionYIndex]);
+    Rotation2d botYaw = new Rotation2d(botpose[botRotationIndex]);
+    Pose2d RealBotpos = new Pose2d(botTranslation, botYaw);
```

Comment:
realRobotPosition is more appropriate if we want to follow naming conventions ... and we do ;)

Reply:
changed